server.go

```diff
@@ -730,7 +749,7 @@ func (sp *serverPeer) OnVersion(_ *peer.Peer, msg *wire.MsgVersion) {
 	}
 
 	// Reject peers that have a protocol version that is too old.
-	if msg.ProtocolVersion < int32(wire.SendHeadersVersion) {
+	if msgProtocolVersion < wire.SendHeadersVersion {
```

Comment:
This is changing the semantics improperly and could lead to a bunch of undesirable things.

Currently, a negative protocol version will be rejected due to this check, but with this change, negative protocol versions would become large positive versions and thus improperly pass this check.

This must be reverted.

The following code demonstrates:

```Go
package main

import "fmt"

func main() {
	var pver int32 = -1
	var testPver uint32 = 3
	if pver < int32(testPver) {
		fmt.Println("original too low as expected")
	}
	if uint32(pver) < testPver {
		fmt.Println("notice the modified version does not see it as too low")
	} else {
		fmt.Println("treated as ok even though it's not")
	}
}
```
Output:
```
original too low as expected
treated as ok even though it's not
````